### PR TITLE
Ci automated js ctests

### PR DESCRIPTION
**Pull request by IonutMuthi**

### CODE REVIEW

File: JS/scopyDefaultTests.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+importFileContent("../JS/scpyDefaultJsFunctions.js")
+
+function connectionTest(){
+    printToConsole("TEST CONNECTION")
+    connect()
+    disconnect()
+}
+
+function switchToolTest(){
+  printToConsole("TEST SWITHC TOOL")
```
Comment (adisuciu) [inline JS/scopyDefaultTests.js:11]: typo

File: JS/scopyDefaultTests.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//This import paht is relative to where scopy is build 
+importFileContent("../JS/scpyDefaultJsFunctions.js")
+
```
Comment (adisuciu) [inline JS/scopyDefaultTests.js:3]: typo

File: JS/scpyDefaultJsFunctions.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+function connect(){
+    //TEST WITH EMU
+    var  deviceID = scopy.addDevice("", "ip:127.0.0.0")
```
Comment (adisuciu) [inline JS/scpyDefaultJsFunctions.js:3]: rename this to common.js or scopycommon.js

File: pluginbase/src/scopyjs.cpp [hunk 4]
```diff
@@ -233,4 +233,56 @@ void ScopyJS::jsCategoryFilter(QLoggingCategory *category)
...
+	if(val.isError()) {
+		out << "Exception:" << val.toString() << Qt::endl;
+	} else if(!val.isUndefined()) {
+		out << val.toString() << Qt::endl;
+	}
+
+	out.flush();
+}
+
+QJSValue ScopyJS::importFileAs(const QString &path)
```
Comment (adisuciu) [inline pluginbase/src/scopyjs.cpp:255]: Do you have an example for this ?

Comment (adisuciu) [reply pluginbase/src/scopyjs.cpp:255]: I think both functions serve the same purpose .. are both needed as one just discards the return value ? you can simply not use it.

Also you can rename this to "evaluateFile"

File: JS/scopyDefaultTests.js (hunk 2 above)
Comment (adisuciu) [reply JS/scopyDefaultTests.js:3]: rename JS to js so it complies with the rest of the project

**Review by adisuciu — CHANGES_REQUESTED** [review]

File: js/scpycommon.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+function connect(){
```
Comment (adisuciu) [inline js/scpycommon.js:1]: typo: file name

**Review by adisuciu — CHANGES_REQUESTED** [review]

File: js/test.sh [hunk 6]
```diff
@@ -0,0 +1,29 @@
+#! /bin/bash 
+# set -xe
+# get scopy buil path from CMakeCache.txt
```
Comment (AlexandraTrifan) [inline js/test.sh:3]: "build" typo

File: plugins/datalogger/js/dataLoggerAutomatedTest.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
+//This import paht is relative to where scopy is build
```
Comment (AlexandraTrifan) [inline plugins/datalogger/js/dataLoggerAutomatedTest.js:1]: path typo

File: plugins/datalogger/js/dataLoggerAutomatedTest.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
+//This import paht is relative to where scopy is build 
+evaluateFile("../js/scpyDefaultJsFunctions.js")
```
Comment (AlexandraTrifan) [inline plugins/datalogger/js/dataLoggerAutomatedTest.js:2]: file name typo

File: tests/CMakeLists.txt [hunk 9]
```diff
@@ -1 +1,26 @@
...
+# JS AUTOMATED TESTS
+option(ENABLE_AUTOMATED_TESTS "Enable JS Automated Tests" OFF)
+
+if(ENABLE_AUTOMATED_TESTS)
+	# Set build path as CMAKE CACHE variable to use in tests
+	set(SCOPY_BUILD_PATH ${CMAKE_BINARY_DIR} CACHE STRING "SCOPY_BUILD_PATH" FORCE)
+
+	# SCOPY GENERIC TESTS
+	add_test(NAME "ScopyDefaultJSTests"
+		 COMMAND bash ${CMAKE_SOURCE_DIR} / js / test.sh "${CMAKE_SOURCE_DIR}/resources/emuXml/"
```
Comment (AlexandraTrifan) [inline tests/CMakeLists.txt:12]: "${CMAKE_SOURCE_DIR} / js / test.sh" are these white spaces intentionally added?

**Review by Andrei-Fabian-Pop — COMMENTED** [review]
Could you also check the indentation in the js and bash scripts, there are some misaligned rows and extra spaces

**Pull request merged**